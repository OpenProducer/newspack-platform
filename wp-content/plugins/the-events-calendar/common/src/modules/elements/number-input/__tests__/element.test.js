/**
 * External dependencies
 */
import React from 'react';
import { noop } from 'lodash';

/**
 * Internal dependencies
 */
import NumberInput from '../element.js';

describe( 'Input element', () => {
	it( 'Should render the component', () => {
		const component = renderer.create( <NumberInput /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	it( 'Should render the component with class', () => {
		const component = renderer.create( <NumberInput className="input-class" /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	it( 'Should render the component with max', () => {
		const component = renderer.create( <NumberInput max={ 42 } /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	it( 'Should render the component with min', () => {
		const component = renderer.create( <NumberInput min={ -42 } /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	it( 'Should render the component with onChange handler', () => {
		const component = renderer.create( <NumberInput onChange={ noop } /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	it( 'Should render the component with step', () => {
		const component = renderer.create( <NumberInput step={ 10 } /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	it( 'Should render the component with extra props', () => {
		const component = renderer.create( <NumberInput id="input-id" /> );
		expect( component.toJSON() ).toMatchSnapshot();
	} );
} );
