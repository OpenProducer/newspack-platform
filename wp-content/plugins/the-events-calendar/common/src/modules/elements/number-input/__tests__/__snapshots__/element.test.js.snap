// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input element Should render the component 1`] = `
<input
  className="tribe-editor__input tribe-editor__input--number"
  type="number"
/>
`;

exports[`Input element Should render the component with class 1`] = `
<input
  className="tribe-editor__input tribe-editor__input--number input-class"
  type="number"
/>
`;

exports[`Input element Should render the component with extra props 1`] = `
<input
  className="tribe-editor__input tribe-editor__input--number"
  id="input-id"
  type="number"
/>
`;

exports[`Input element Should render the component with max 1`] = `
<input
  className="tribe-editor__input tribe-editor__input--number"
  max={42}
  type="number"
/>
`;

exports[`Input element Should render the component with min 1`] = `
<input
  className="tribe-editor__input tribe-editor__input--number"
  min={-42}
  type="number"
/>
`;

exports[`Input element Should render the component with onChange handler 1`] = `
<input
  className="tribe-editor__input tribe-editor__input--number"
  onChange={[Function]}
  type="number"
/>
`;

exports[`Input element Should render the component with step 1`] = `
<input
  className="tribe-editor__input tribe-editor__input--number"
  step={10}
  type="number"
/>
`;
